common-lisp-parachute/source/gilded-rose.lisp

```diff
@@ -0,0 +1,128 @@
+;; "Conjured" items degrade in Quality twice as fast as normal items
+;; Feel free to make any changes to the UpdateQuality method and add
+;; any new code as long as everything still works correctly. However,
+;; do not alter the Item class or Items property as those belong to the
+;; goblin in the corner who will insta-rage and one-shot you as he
+;; doesn't believe in shared code ownership (you can make the
+;; UpdateQuality method and Items property static if you like, we'll
+;; cover for you).
+;; Just for clarification, an item can never have its Quality increase
+;; above 50, however "Sulfuras" is a legendary item and as such its
+;; Quality is 80 and it never alters.
+
+;; https://github.com/emilybache/GildedRose-Refactoring-Kata
+
+;; Common Lisp version: Rainer Joswig, joswig@lisp.de, 2016
+;; Minor modifications (independent of CL impl.): Manfred Bergmann, 2022
+;; Adaption to asdf: Nico Simoski, 2023
+
+;;; ================================================================
+;;; Code
+
+(in-package :gilded-rose)
+
+;;; Class ITEM
+
+(defclass item ()
+  ((name    :initarg :name    :type string)
+   (sell-in :initarg :sell-in :type integer)
+   (quality :initarg :quality :type integer)))
+
+(defmethod to-string ((i item))
+  (with-slots (name quality sell-in) i
+    (format nil "~a, ~a, ~a" name sell-in quality)))
+
+;;; Class gilded-rose
+
+(defclass gilded-rose ()
+  ((items :initarg :items)))
+
+(defmethod update-quality ((gr gilded-rose))
+  (with-slots (items) gr
+    (dotimes (i (length items))
+      (with-slots (name quality sell-in)
+          (elt items i)
+        (if (and (not (equalp name "Aged Brie"))
+                 (not (equalp name "Backstage passes to a TAFKAL80ETC concert")))
+            (if (> quality 0)
+                (if (not (equalp name "Sulfuras, Hand of Ragnaros"))
+                    (setf quality (- quality 1))))
+          (when (< quality 50)
+            (setf quality (+ quality 1))
+            (when (equalp name "Backstage passes to a TAFKAL80ETC concert")
+              (if (< sell-in 11)
+                  (if (< quality 50)
+                      (setf quality (+ quality 1))))
+              (if (< sell-in 6)
+                  (if (< quality 50)
+                      (setf quality (+ quality 1)))))))
+
+        (if (not (equalp name "Sulfuras, Hand of Ragnaros"))
+            (setf sell-in (- sell-in 1)))
+
+        (if (< sell-in 0)
+            (if (not (equalp name "Aged Brie"))
+                (if (not (equalp name "Backstage passes to a TAFKAL80ETC concert"))
+                    (if (> quality 0)
+                        (if (not (equalp name "Sulfuras, Hand of Ragnaros"))
+                            (setf quality (- quality 1))))
+                  (setf quality (- quality quality)))
+              (if (< quality 50)
+                  (setf quality (+ quality 1)))))))))
+
+;;; Example
+
+(defun run-gilded-rose (days)
```

Comment:
Can you include in the README a description of how to execute this function as a command-line program? That's the 'texttest fixture'